File: archive/c/c-sharp/Rot13.cs
if (args is not [var input] || string.IsNullOrEmpty(input))
    return ExitWithUsage();

Console.WriteLine(Rot13(input.AsSpan()));
return 0;

static string Rot13(ReadOnlySpan<char> input)
{
    char[] result = new char[input.Length];

    for (int i = 0; i < input.Length; i++)
    {
        char c = input[i];
        result[i] = c switch
        {
            >= 'a' and <= 'z' => (char)('a' + (c - 'a' + 13) % 26),
            >= 'A' and <= 'Z' => (char)('A' + (c - 'A' + 13) % 26),
            _ => c,
        };
    }

    return new string(result);
}

static int ExitWithUsage()
{
    Console.WriteLine("Usage: please provide a string to encrypt");
    return 1;
}

File: archive/v/visual-basic/rot13.vb
Imports System.Text

Public Module Rot13

    Private Sub ShowUsage()
        Console.WriteLine("Usage: please provide a string to encrypt")
        Environment.Exit(1)
    End Sub


    Private Function EncryptChar(c As Char) As Char

        If c >= "A"c AndAlso c <= "Z"c Then
            Return ChrW(((AscW(c) - AscW("A"c) + 13) Mod 26) + AscW("A"c))
        End If

        If c >= "a"c AndAlso c <= "z"c Then
            Return ChrW(((AscW(c) - AscW("a"c) + 13) Mod 26) + AscW("a"c))
        End If

        Return c

    End Function


    Private Function Encrypt(input As String) As String

        Dim result As New StringBuilder(input.Length)

        For Each c As Char In input
            result.Append(EncryptChar(c))
        Next

        Return result.ToString()

    End Function


    Public Function Main(args As String()) As Integer

        If args Is Nothing OrElse args.Length = 0 OrElse String.IsNullOrEmpty(args(0)) Then
            ShowUsage()
        End If

        Console.WriteLine(Encrypt(args(0)))

        Return 0

    End Function

End Module